taichi/transforms/lower_ast.cpp

```diff
@@ -121,21 +134,21 @@ class LowerAST : public IRVisitor {
     // transform into a structure as
     // while (1) { cond; if (no active) break; original body...}
     auto cond = stmt->cond;
-    VecStatement flattened;
-    cond->flatten(flattened);
-    auto cond_stmt = flattened.back().get();
+    auto fctx = make_flatten_ctx();
+    cond->flatten(&fctx);
+    auto cond_stmt = fctx.back_stmt();
 
     auto &&new_while = std::make_unique<WhileStmt>(std::move(stmt->body));
     auto mask = std::make_unique<AllocaStmt>(DataType::i32);
     new_while->mask = mask.get();
     auto &stmts = new_while->body;
-    for (int i = 0; i < (int)flattened.size(); i++) {
-      stmts->insert(std::move(flattened[i]), i);
+    for (int i = 0; i < (int)fctx.stmts.size(); i++) {
+      stmts->insert(std::move(fctx.stmts[i]), i);
```

Comment:
It looks the same as `stmts->insert(std::move(fctx.stmts), 0)`. Although this is probably not directly related to this PR :)

Reply:
Nice thanks!